src/main/java/org/opensearch/knn/index/codec/KNNCodecService.java

```diff
@@ -27,6 +26,6 @@ public KNNCodecService(CodecServiceConfig codecServiceConfig) {
      */
     @Override
     public Codec codec(String name) {
-        return new KNN87Codec(super.codec(name));
+        return KNNCodecFactory.createKNN91Codec(super.codec(name));
```

Comment:
Did you consider making 2 methods for KNNCodecFactory:
```
KNNCodecFactory.createKNNCodec(string knnCodecVersion, codec delegate)

// Just returns the latest
KNNCodecFactory.createKNNCodec(codec delegate)

```

That way, we could avoid having to change this class during upgrade.

Reply:
Good idea, will add this in next commit